presto-main/src/main/java/com/facebook/presto/operator/aggregation/histogram/GroupedHistogramState.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.operator.aggregation.histogram;
+
+import com.facebook.presto.operator.aggregation.state.AbstractGroupedAccumulatorState;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.type.Type;
+import org.openjdk.jol.info.ClassLayout;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static it.unimi.dsi.fastutil.HashCommon.arraySize;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * state object that uses a single histogram for all groups. See {@link GroupedTypedHistogram}
+ */
+public class GroupedHistogramState
+        extends AbstractGroupedAccumulatorState
+        implements HistogramState
+{
+    private static final int INSTANCE_SIZE = ClassLayout.parseClass(GroupedHistogramState.class).instanceSize();
+    private TypedHistogram typedHistogram;
+    private long size;
+    private BlockBuilder valuesBlockBuilder;
+
+    public GroupedHistogramState(Type keyType, int expectedEntriesCount)
+    {
+        valuesBlockBuilder = keyType.createBlockBuilder(null, computeBucketCount(expectedEntriesCount, GroupedTypedHistogram.MAX_FILL_RATIO));
```

Comment:
Why does this class need to know about values BlockBuilder? It seems like it should be an implementation detail of GroupedTypeHistogram (which already takes the keyType, expectedEntriesCount as arguments and has an internal constant for MAX_FILL_RATIO)

Reply:
good point--i think this is a holdover from my thinking in the first cut that we've removed--create a shared "value builder" and pass it into multiple instances  of the GroupedTypedHistogram

I can move this into the class
will update

Reply:
fixed